### PR TITLE
chore: update yauzl-promise type definitions for v4.0

### PR DESCRIPTION
**Pull request by RedlineTriad**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  https://github.com/overlookmotel/yauzl-promise/blob/master/README.md?plain=1#L18-L22
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/yauzl-promise/yauzl-promise-tests.ts
1. types/yauzl-promise/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 71c20fe20a9d7fc26abc2e4cc7d8405f7d539520
</p>

File: types/yauzl-promise/package.json [hunk 1]
```diff
@@ -1,13 +1,12 @@
 {
     "private": true,
     "name": "@types/yauzl-promise",
-    "version": "2.1.9999",
+    "version": "3.0.9999",
```
Comment (peterblazejewicz) [inline types/yauzl-promise/package.json:4]: the title says '4', but here the version is v3.

Comment (RedlineTriad) [reply types/yauzl-promise/package.json:4]: The newest version of yauzl-promise is v4, but I thought I should just bump the @types dependency by a major version.
Should I try to match yauzl-promises version?

Comment (peterblazejewicz) [reply types/yauzl-promise/package.json:4]: Yes, please, there is no need for version consistency here. With DT, latest available is matched

Comment (RedlineTriad) [reply types/yauzl-promise/package.json]: Is now as close as I can get to upstream version

**Review by peterblazejewicz — APPROVED** [review]
@RedlineTriad thanks!

**Comment by RedlineTriad** [comment]
Ready to merge

**Pull request merged**